.github/workflows/release.yaml

```diff
@@ -14,6 +14,8 @@ jobs:
   release:
     name: Release
     runs-on: ubuntu-latest
+    permissions:
+      contents: write  # Required to upload release assets
```

Comment:
```suggestion
```

It already have this?
<https://github.com/home-assistant/frontend/runs/5578172811?check_suite_focus=true#step:1:16>

Reply:
It's recommended to limit the permissions to what's necessary.
https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token
```
As a good security practice, you should grant the GITHUB_TOKEN the least required access.
```

By adding only `contents: write` explicitly, I exclude all other permission scopes.

Reply:
If that is the reason behind it, it should be defined on the workflow as read, and then open each job to what they need.

Reply:
https://github.com/home-assistant/frontend/pull/11566/commits/31cc0a1a6f65dfdaece6edc0f5d983af0f48f5e9 I checked the other jobs. None need additional permissions.
_I wasn't sure about artifact upload / download, but after testing it seems to work just fine._

For the future, it might be worth considering to limit the default repo[^1] / org[^2] permissions for Github tokens.
With regards to the other workflows, only a few would need explicit permissions.
* `release-drafter` -> `contents: write` to create pre-releases
* `lock`[^3] and `stale`[^4] -> `issues: write` and `pull-requests: write`

[^1]: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#setting-the-permissions-of-the-github_token-for-your-repository
[^2]: https://docs.github.com/en/organizations/managing-organization-settings/disabling-or-limiting-github-actions-for-your-organization#setting-the-permissions-of-the-github_token-for-your-organization
[^3]: https://github.com/dessant/lock-threads#examples
[^4]: https://github.com/actions/stale#recommended-permissions

Reply:
Action artifacts needed "deploy" before, but if it works without anything that's just great 👍 
As for changing the default of the entire org, I 100% agree with that, both in regards to the token, and potentially implementing an allow list for approved actions.